xlators/protocol/client/src/client.c

```diff
@@ -796,7 +796,8 @@ client_open(call_frame_t *frame, xlator_t *this, loc_t *loc, int32_t flags,
         pthread_spin_lock(&conf->fd_lock);
         {
             fdctx = this_fd_get_ctx(fd, this);
-            if (fdctx && !list_empty(&fdctx->lock_list)) {
+            if (fdctx && !(list_empty(&fdctx->lock_list) &&
+                           fd_lk_ctx_empty(fdctx->lk_ctx))) {
```

Comment:
Please make this into a function and remove duplication of logic

Reply:
Done.